iroh-resolver/src/resolver.rs

```diff
@@ -581,6 +617,42 @@ impl<T: ContentLoader> Resolver<T> {
         }
     }
 
+    /// Resolve a path recursively and yield the raw bytes plus metadata.
+    #[tracing::instrument(skip(self))]
+    pub fn resolve_recursive_raw(&self, root: Path) -> impl Stream<Item = Result<OutRaw>> {
+        let mut cids = VecDeque::new();
+        let this = self.clone();
+        async_stream::try_stream! {
+            let root_block = this.resolve_root(&root).await.map(|(cid, loaded)| OutRaw::from_loaded(cid, loaded));
```

Comment:
I think this is the same code as `resolve_recursive` effectively, other than mapping things differently, maybe this could be abstracted by passing a closer to a shared function?

Reply:
Sounds good. I implemented an abstraction to be shared between the two `resolve_recursive` functions.  `resolve_recursive_with_path` doesn't yet use it because there's additional logic around keeping the path. Could add it to the generic one, if always maintaining the path isn't considered too expensive (don't think it would matter).

Reply:
@dignifiedquire I've looked into having the remaining `resolve_recursive_with_path` also use the `resolve_recursive_mapped` function from the latest commit in this PR. I made it work but with the cost of a couple of additional clones of the path even if it's not being used. See this commit (not on this PR): https://github.com/Frando/iroh/commit/91c04401c27ea593cd7cdd24a44def2186f7ce0d
I wasn't sure if it's worth it.